### PR TITLE
fix(cli): handle run mode correctly (fix #719)

### LINKED ISSUE #719: vitest run is watching files and does not quit
### Describe the bug

`vitest run` is now watching files.

image

### Reproduction

Run `vitest run`

### System Info

```shell
System:
    OS: macOS 12.1
    CPU: (8) arm64 Apple M1
    Memory: 231.05 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - /var/folders/9v/sx18bwn925b3f2rp2jfg952w0000gn/T/fnm_multishells/13654_1644417845070/bin/node
    Yarn: 1.22.15 - /var/folders/9v/sx18bwn925b3f2rp2jfg952w0000gn/T/fnm_multishells/13654_1644417845070/bin/yarn
    npm: 8.3.0 - /var/folders/9v/sx18bwn925b3f2rp2jfg952w0000gn/T/fnm_multishells/13654_1644417845070/bin/npm
  Browsers:
    Chrome: 98.0.4758.80
    Safari: 15.2
  npmPackages:
    vite: ^2.8.0 => 2.8.0
    vitest: ^0.3.0 => 0.3.0
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by DerYeger**
Closes #719

### CODE REVIEW

File: packages/vitest/src/constants.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export const configDefaults: UserConfig = Object.freeze({
   isolate: true,
   watchIgnore: [/\/node_modules\//, /\/dist\//],
   update: false,
+  run: !!process.env.CI,
```
Comment (antfu) [inline packages/vitest/src/constants.ts:52]: The fix should be make watch default to false and handles it in CLI, /cc @sheremet-va

Comment (DerYeger) [reply packages/vitest/src/constants.ts:52]: Should be fixed

File: packages/vitest/src/node/cli.ts [hunk 2]
```diff
@@ -35,34 +35,34 @@ cli
...
 
 cli
   .command('dev [...filters]')
-  .action(dev)
+  .action(run)
 
 cli
   .command('[...filters]')
-  .action(dev)
+  .action(run)
```
Comment (antfu) [inline packages/vitest/src/node/cli.ts:54]: Can we keep the original naming? I found them a bit confusing

File: packages/vitest/src/node/plugins/index.ts [hunk 3]
```diff
@@ -56,7 +56,7 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
           options,
         )
         options.api = resolveApiConfig(options)
-        options.watch = options.watch && !options.run
+        options.watch = !options.run
```
Comment (antfu) [inline packages/vitest/src/node/plugins/index.ts:59]: Should keep the original logic.

There are some logic under `dev` get removed in #702, we need to bring them back

Comment (DerYeger) [reply packages/vitest/src/node/plugins/index.ts]: The logic of the `dev` function that got removed in #702 isn't needed anymore.
It was just acting as a default value, which is now included in `configDefaults`.

**Review by patak-dev — APPROVED** [review]
The fix looks good after the discussion. It would be good to avoid the redundant booleans `{ run: true, watch: false }`, maybe a mode could be explored? A `mode: 'watch' | 'run'`, like we have in vite with `'dev' | 'build'`.

**Comment by DerYeger** [comment]
> The fix looks good after the discussion. It would be good to avoid the redundant booleans `{ run: true, watch: false }`, maybe a mode could be explored? A `mode: 'watch' | 'run'`, like we have in vite with `'dev' | 'build'`.

Absolutely, the separate `run` and `watch` are cumbersome.
We should use a mode type in our config and just parse the CLI args accordingly.
Since the current situation is quite a big issue I'd prefer to do it in a separate PR though.

**Pull request closed without merging**